File: CivitasJava/src/civitas/SorpresaPagarCobrar.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author luis
 */
public class SorpresaPagarCobrar extends Sorpresa{
    
    private int valor;
    
    SorpresaPagarCobrar(int valor){
        super("SorpresaPagarCobrar");
        this.valor = valor;
    }
    
    @Override
    public void aplicarAJugador(int actual, ArrayList<Jugador> todos){
        if(jugadorCorrecto(actual,todos)){
            informe(actual,todos);
            todos.get(actual).modificarSaldo(this.valor);
        }
    }
    
    @Override
    public String toString(){
        String e = "\nNombre: SorpresaPagarCobrar";
        return e;
    }
}

File: CivitasRuby/lib/sorpresa_pagar_cobrar.rb
# encoding:utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.


module Civitas
  class SorpresaPagarCobrar < Sorpresa
    
    def initialize(valor)
       super("SorpresaPagarCobrar")
       @valor = valor
    end
    
    
    def aplicar_a_jugador(actual, todos)
      if jugador_correcto(actual,todos)
        informe(actual,todos)
        todos[actual].modificar_saldo(@valor)
      end
    end
    
    def to_s
      "\nNombre: #{@texto}"
    end
    
       
    public_class_method  :new
    
  end
end
